modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/Attachments.tsx

```diff
@@ -90,20 +90,28 @@ export function Attachments({setValues, values}: IProps) {
 
 		const {items} = (await response.json()) as {items: ObjectField[]};
 
-		const fields: ObjectField[] = items.filter(
+		const fields: ObjectField[] = items?.filter(
 			(field) => field.businessType === 'Attachment'
 		);
 
 		setAttachmentsFields(parseFields(fields));
 	};
 
 	useEffect(() => {
-		setSelectedEntity(
-			objectDefinitions?.find(
-				(item) => item.id === values.objectDefinitionId
-			)
+		const currentObjectDefinition = objectDefinitions?.find(
+			(item) => item.id === values.objectDefinitionId
 		);
 
+		if (!currentObjectDefinition) {
+			setValues({
+				...values,
+				attachmentObjectFieldIds: [],
+				objectDefinitionId: null,
+			});
+		}
+
+		setSelectedEntity(currentObjectDefinition);
```

Comment:
There is a possibility of settings this as undefined, is this really okay?

Reply:
Yes. If selectedEntity is `undefined` it will reset the dropdown to the empty option which is `Select Data Source`.

Reply:
Oh I see! I understand now, so I will forward this PR. Thanks! :heart: